src/queries/utils.ts

```diff
@@ -0,0 +1,39 @@
+import {
+  DatabaseTransactionConnectionType,
+  TaggedTemplateLiteralInvocationType,
+  sql,
+} from 'slonik';
+import { EitherAsync } from 'purify-ts';
+
+export type DateTime = Date;
+
+export type SnakeToCamel<S extends string> = S extends `${infer T}_${infer U}`
+  ? `${T}${Capitalize<SnakeToCamel<U>>}`
+  : S;
+
+export type CamelToSnake<S extends string> = S extends `${infer T}${infer U}`
+  ? `${T extends Capitalize<T> ? '_' : ''}${Lowercase<T>}${CamelToSnake<U>}`
+  : S;
+
+export const select = (columns: string[]) =>
+  sql.join(
+    columns.map(column => sql.identifier([column])),
+    sql`, `,
+  );
+
+export const camelToSnakeCase = (str: string) =>
+  str.replace(/[A-Z]/g, letter => `_${letter.toLowerCase()}`);
+
+export function fetchOne<I, O>(action: (i: I) => TaggedTemplateLiteralInvocationType<O>) {
```

Comment:
Could these be generalized a step further? As it is, we need a function for each slonik query method. Maybe pass `trx` to action and let it decide which query method to use.

Reply:
I've been thinking of how to do that, but what you suggested might just work!

Reply:
Done, seems to have worked quite well 👍